python/perftest/models/v1alpha1/pytorch.py

```diff
@@ -0,0 +1,283 @@
+    )
+        
+
+class PytorchResult(schema.BaseModel):
+    """
+    Represents an individual Pytorch benchmark result.
+    
+    Some notes on the inner workings of the pytorch benchmark script:
+    - Currently only runs one batch for benchmark so 'time per batch' in pytorch output == total time.
+      (This may change in future since 'per batch' suffix was added to output text very recently.)
+      https://github.com/pytorch/benchmark/blob/6fef32ddaf93a63088b97eb27620fb57ef247521/run.py#L468
+    - CPU 'wall time' reported by pytorch is significantly shorter than reported by GNU `time` command.
+      It's not clear what is taking up this extra time outwith the actual model invocation (downloading
+      model weights and generating random in-memory input data shouldn't take long at all).
+    """
+    pytorch_time: schema.confloat(ge = 0) = Field(
+        ...,
+        description = "The CPU wall time (in seconds) as reported by the pytorch benchmark script."
+    )
+    peak_cpu_memory_GB: schema.confloat(ge = 0) = Field(
+        ...,
+        description = "The peak CPU memory usage (in GB) reported by the pytorch benchmark script."
+    )
+    gpu_time: t.Optional[schema.confloat(ge = 0)] = Field(
+        None, # Default to zero for clearer reporting on cpu-only runs
+        description = "The GPU wall time (in seconds) reported by the pytorch benchmark script."
+    )
+    peak_gpu_memory_GB: t.Optional[schema.confloat(ge = 0)] = Field(
+        None, # Default to zero for clearer reporting on cpu-only runs
+        description = "The peak GPU memory usage (in GB) reported by the pytorch benchmark script."
+    )
+    gnu_time: GnuTimeResult = Field(
+        description = "A container for the output of the `time` command which wraps the benchmark execution script."
+    )
+    
+
+class PytorchStatus(base.BenchmarkStatus):
+    """
+    Represents the status of the Pytorch benchmark.
+    """
+    gpu_count: conint(ge=0) = Field(
+        None,
+        description = "The number of gpus used in this benchmark"
+    )
+    result: t.Optional[PytorchResult] = Field(
+        None,
+        description = "The result of the benchmark."
+    )
+    wall_time_result: schema.confloat(ge = 0) = Field(
+        None,
+        description = (
+            "The wall time (in seconds) reported by the GNU time wrapper."
+            "Used as a headline result."
+        )
+    )
+    gpu_time_result: schema.confloat(ge = 0) = Field(
+        None,
+        description = (
+            "The GPU wall time (in seconds) reported by the pytorch benchmark script."
+            "Used as a headline result."
+        )
+    )
+    worker_pod: t.Optional[base.PodInfo] = Field(
+        None,
+        description = "Pod information for the pod running the benchmark."
+    )
+    client_log: t.Optional[constr(min_length = 1)] = Field(
+        None,
+        description = "The raw pod log of the client pod."
+    )
+
+
+class Pytorch(
```

Comment:
The proper capitalisation is `PyTorch` - we should change that for all the class names in here.